app/src/components/home/Home.test.js

```diff
@@ -0,0 +1,123 @@
+import { MemoryRouter } from 'react-router-dom';
+import { mount, shallow } from 'enzyme';
+
+import configureStore from '../../store/configureStore';
+import Home from './Home';
+//next import will load envVars from local override of app/public/conf.js
+import '../../../public/conf';
+
+const div = document.createElement('div');
+let store;
+beforeEach(() => {
+  store = configureStore();
+});
+
+it('deep renders home component', () => {
+  render(
+    <Provider store={store}>
+      <MemoryRouter>
+        <Home />
+      </MemoryRouter>
+    </Provider>, div);
+  unmountComponentAtNode(div);
+});
+
+it('has one input element', () => {
+  const wrapper = mount(
+    <Provider store={store}>
+      <MemoryRouter>
+        <Home />
+      </MemoryRouter>
+    </Provider>);
+  expect(wrapper.find('input')).toHaveLength(1);
+});
+
+it('submits a valid query', () => {
+  const SEARCH_STR = 'foobar';
+  const wrapper = mount(
+    <Provider store={store}>
+      <MemoryRouter>
+        <Home />
+      </MemoryRouter>
+    </Provider>);
+
+  //change search text in menu bar input
+  let input = wrapper.find('input').first();
+  input.simulate('change', { target: { value: SEARCH_STR}});
+  //and submit the query
+  input.simulate('keypress', {key: 'Enter'});
+  //check for expected state of the store
+  const nowStore = store.getState();
+  expect(nowStore.query.current).toEqual(SEARCH_STR);
+  expect(nowStore.query.submitted).toEqual(true);
+  expect(nowStore.query.isWaiting).toEqual(true);
+
+});
+
+it('tries to submit an empty query', () => {
+  const SEARCH_STR = '';
+  const wrapper = mount(
+    <Provider store={store}>
+      <MemoryRouter>
+        <Home />
+      </MemoryRouter>
+    </Provider>);
+
+  //change search text in menu bar input
+  let input = wrapper.find('input').first();
+  input.simulate('change', { target: { value: SEARCH_STR}});
+  //and try to submit the query
+  input.simulate('keypress', {key: 'Enter'});
+  //check for expected state of the store
+  const nowStore = store.getState();
+  expect(nowStore.query.current).toEqual(SEARCH_STR);
+  expect(nowStore.query.submitted).toEqual(false);
+  expect(nowStore.query.isWaiting).toEqual(false);
+});
+
+it('tries to submit too short a query', () => {
+  const SEARCH_STR = 'fo';
+  const wrapper = mount(
+    <Provider store={store}>
+      <MemoryRouter>
+        <Home />
+      </MemoryRouter>
+    </Provider>);
+
+  //change search text in menu bar input
+  let input = wrapper.find('input').first();
+  input.simulate('change', { target: { value: SEARCH_STR}});
+  //and try to submit the query
+  input.simulate('keypress', {key: 'Enter'});
+  //check for expected state of the store
+  const nowStore = store.getState();
+  expect(nowStore.query.current).toEqual(SEARCH_STR);
+  expect(nowStore.query.submitted).toEqual(false);
+  expect(nowStore.query.isWaiting).toEqual(false);
+  //also check for existience of Notification
```

Comment:
Is there a specific way to check the notification here ?

Reply:
yes, there is a way but it's not implemented yet

really, that check may be better as a separate test (one checking data, the other looking for DOM node creation)

I'll mark that comment as a TODO to come back to soon and will drop a few of my quick thoughts there, as well